packages/components/src/internal/components/administration/GroupAssignments.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { ChangeEvent, FC, memo, useCallback, useMemo, useState } from 'react';
+import { Button, Col, Panel, Row } from 'react-bootstrap';
+
+import { List, Map } from 'immutable';
+
+import { Alert } from '../base/Alert';
+
+import { Principal, SecurityPolicy, SecurityRole } from '../permissions/models';
+import { UserDetailsPanel } from '../user/UserDetailsPanel';
+import { GroupDetailsPanel } from '../permissions/GroupDetailsPanel';
+
+import { Group } from './Group';
+import { GroupMembership } from './models';
+
+export interface GroupAssignmentsProps {
+    addMembers: (groupId: string, principalId: number, principalName: string, principalType: string) => void;
+    createGroup: (name: string) => void;
+    deleteGroup: (id: string) => void;
+    errorMsg: string;
+    groupMembership: GroupMembership;
+    policy: SecurityPolicy;
+    principalsById: Map<number, Principal>;
+    removeMember: (groupId: string, memberId: number) => void;
+    rolesByUniqueName: Map<string, SecurityRole>;
+    save: () => Promise<void>;
+    setErrorMsg: (e: string) => void;
+    showDetailsPanel?: boolean;
+    usersAndGroups: List<Principal>;
+}
+
+export const GroupAssignments: FC<GroupAssignmentsProps> = memo(props => {
+    const {
+        errorMsg,
+        groupMembership,
+        showDetailsPanel = true,
+        policy,
+        rolesByUniqueName,
+        principalsById,
+        usersAndGroups,
+        createGroup,
+        deleteGroup,
+        addMembers,
+        removeMember,
+        save,
+        setErrorMsg,
+    } = props;
+
+    const [dirty, setDirty] = useState<boolean>();
+    const [submitting, setSubmitting] = useState<boolean>(false);
+    const [selectedPrincipalId, setSelectedPrincipalId] = useState<number>();
+    const [newGroupName, setNewGroupName] = useState<string>('');
+
+    const onSave = useCallback(async () => {
+        setSubmitting(true);
+        setErrorMsg(undefined);
+        await save();
```

Comment:
Since the save() fn itself has a try-catch, it should end up caught in the save on the GroupManagement.tsx page. Tried this out with setting the throttling to offline and works as expected!

Reply:
Ah yes, I didn't catch that during my review.